BotrAPI.php

```diff
@@ -97,57 +203,82 @@ private function _args($args)
         return $args;
     }
 
-    // Construct call URL
-    public function call_url($call, $args=array())
+    /**
+     * Gets the full API URL to use for a call
+     *
+     * @param string $call the API call to make.
+     * @param array  $args optional. The call arguments.
+     *
+     * @return string the full API URL to use.
+     */
+    protected function getCallURL($call, array $args = array())
     {
         $url = $this->_url . $call . '?' . http_build_query($this->_args($args), "", "&");
         return $url;
     }
 
-    // Make an API call
-    public function call($call, $args=array())
+    /**
+     * Makes an API call
+     *
+     * @param string $call the API call to make.
+     * @param array  $args optional. The call arguments.
+     *
+     * @return array the response object.
+     */
+    public function call($call, array $args = array())
     {
-        $url = $this->call_url($call, $args);
+        $url = $this->getCallURL($call, $args);
 
         $response = null;
         switch($this->_library) {
-            case 'curl':
-                $curl = curl_init();
-                curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
-                curl_setopt($curl, CURLOPT_URL, $url);
-                $response = curl_exec($curl);
-                break;
-            default:
-                $response = file_get_contents($url);
+        case 'curl':
+            $curl = curl_init();
+            curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+            curl_setopt($curl, CURLOPT_URL, $url);
+            $response = curl_exec($curl);
+            break;
+        default:
+            $response = file_get_contents($url);
+            break;
         }
 
         $unserialized_response = @unserialize($response);
 
         return $unserialized_response ? $unserialized_response : $response;
     }
 
-    // Upload a file
-    public function upload($upload_link=array(), $file_path, $api_format="php")
+    /**
+     * Uploads a file to BotR
+     *
+     * @param array  $upload_link information about the file to upload.
+     * @param string $file_path   path to the file to upload.
+     * @param string $api_format  optional. The requested response format.
+     *                            Defaults to PHP.
+     *
+     * @return array the response object.
+     */
+    public function upload(array $upload_link, $file_path, $api_format = "php")
```

Comment:
Should the $upload_link still be an empty array or is that safety precaution unnecessary?